src/Tools/dotnet-monitor/CollectionRules/Options/CollectionRuleDefaultsPostConfigure/EgressPostConfigure.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Actions;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options.CollectionRuleDefaultsInterfaces
+{
+    internal sealed class EgressPostConfigure :
+        IPostConfigureOptions<CollectionRuleOptions>
+    {
+        private readonly IOptionsMonitor<CollectionRuleDefaultsOptions> _defaultOptions;
+        private readonly ICollectionRuleActionOperations _actionOperations;
+
+        public EgressPostConfigure(
+            IOptionsMonitor<CollectionRuleDefaultsOptions> defaultOptions,
+            ICollectionRuleActionOperations actionOperations
+            )
+        {
+            _defaultOptions = defaultOptions;
+            _actionOperations = actionOperations;
+        }
+
+        public void PostConfigure(string name, CollectionRuleOptions options)
+        {
+            foreach (var action in options.Actions)
+            {
+                _actionOperations.TryCreateOptions(action.Type, out object actionSettings);
+
+                if (null != actionSettings && typeof(IEgressProviderProperties).IsAssignableFrom(actionSettings.GetType()))
```

Comment:
```suggestion
                if (null != actionSettings && actionSettings is IEgressProviderProperties egressProviderProperties)
```

Then you can use `egressProviderProperties` later without casting.

Reply:
Without really investigating this, my tests failed after making this change. My first guess would be that it's an issue related to getting a copy and thus the assignment isn't reflected in the actual value - but I can look more into this since I don't have experience doing assignment with the `is` operator.

Reply:
Oh, nevermind - it was actually that we're only using `actionSettings` for its type and that it's not connected to `action.Settings`. If I set `actionSettings = action.Settings` before we do this check, then it works since then things are properly connected.